services/account/apps/api/src/services/accounts.service.ts

```diff
@@ -40,6 +40,20 @@ export class AccountsService {
     }
   }
 
+  async getMsaIdForPublicKey(publicKey: string): Promise<MsaIdResponse> {
+    try {
+      const msaId = await this.blockchainService.publicKeyToMsaId(publicKey);
+      if (msaId) {
+        this.logger.debug(`Found msaId: ${msaId} for public key: ${publicKey}`);
+        return { msaId };
+      }
+      throw new Error(`Failed to find msaId for public key: ${publicKey}`);
+    } catch (e) {
+      this.logger.error(`Error during get msaId request: ${e}`);
+      throw new Error('Failed to get msaId');
+    }
+  }
+
```

Comment:
Should return `Promise<obj | null>`, and bubble up to controller, which can return a 404 if no account found for the pubkey.